packages/discovery-provider/src/queries/get_trending_playlists.py

```diff
@@ -225,14 +225,21 @@ def wrapped():
         results = []
         for playlist in playlists:
             playlist_owner_id = playlist["playlist_owner_id"]
-            track_owner_ids = list(
-                filter(
-                    lambda owner_id: owner_id != playlist_owner_id,
-                    map(lambda track: track["owner_id"], playlist["tracks"]),
-                )
-            )
-            if len(track_owner_ids) < 3:
+            unique_track_owner_ids = set()
+            valid_track_count = 0
+
+            for track in playlist["tracks"]:
```

Comment:
1. Mind adding a comment for why we're doing this check?
2. My concern with not doing it in the SQL query is that this will filter out results after the LIMIT and we end up with less than the requested amount. My hope was that we could use the filtering mechanism in the query to let it continue to add rows to the result as long as they pass the criteria. We could always request _more_ than we need here as padding. It would still technically have an edge case where we could filter out more than the padding and end up with less than the requested amount of items. But that seems highly unlikely.
3. Also mildly concerned that this doesn't support pagination correctly and it's called by an endpoint that _does_ accept pagination parameters. But I think that was already broken?

Reply:
Ah woops, I had auto merge enabled.

I'll add a comment.

It is no different than how it currently works, so I thought it would be ok.

We clearly don't support proper pagination because we always take a limit of 30 trending things... Someday we will revisit this scoring mechanism wholesale